algorithms/ar-bsrh/typescript/binary_search.ts

```diff
@@ -0,0 +1,23 @@
+function binarySearch(itemList: number[], item: number) {
```

Comment:
Codacy found an issue: [Function 'binarySearch' has a complexity of 5. Maximum allowed is 4.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6274005)